File: training-data/csharp/BuilderPatternDemo.cs
using System;
using System.Collections.Generic;

class Pizza
{
    public List<string> Toppings = new();
    public string Size = "medium";
}

class PizzaBuilder
{
    private readonly Pizza pizza = new();

    public PizzaBuilder WithSize(string size)
    {
        pizza.Size = size;
        return this;
    }

    public PizzaBuilder AddTopping(string topping)
    {
        pizza.Toppings.Add(topping);
        return this;
    }

    public Pizza Build() => pizza;

    static void Main()
    {
        var pizza = new PizzaBuilder().WithSize("large").AddTopping("cheese").AddTopping("olives").Build();
        Console.WriteLine($"{pizza.Size}: {string.Join(", ", pizza.Toppings)}");
    }
}

File: training-data/objectivec/BuilderPatternDemo.m
#import <Foundation/Foundation.h>

@interface Pizza : NSObject
@property (nonatomic, copy) NSString *size;
@property (nonatomic, strong) NSMutableArray<NSString *> *toppings;
- (NSString *)describe;
@end

@implementation Pizza
- (instancetype)init {
    self = [super init];
    if (self) _toppings = [NSMutableArray array];
    return self;
}
- (NSString *)describe {
    return [NSString stringWithFormat:@"%@ pizza with %@", self.size,
            [self.toppings componentsJoinedByString:@", "]];
}
@end

// The builder assembles a Pizza step by step and returns itself from each
// step, so calls can be chained before the final object is produced.
@interface PizzaBuilder : NSObject
- (instancetype)setSize:(NSString *)size;
- (instancetype)addTopping:(NSString *)topping;
- (Pizza *)build;
@end

@implementation PizzaBuilder {
    Pizza *_pizza;
}

- (instancetype)init {
    self = [super init];
    if (self) _pizza = [[Pizza alloc] init];
    return self;
}

- (instancetype)setSize:(NSString *)size {
    _pizza.size = size;
    return self;
}

- (instancetype)addTopping:(NSString *)topping {
    [_pizza.toppings addObject:topping];
    return self;
}

- (Pizza *)build {
    return _pizza;
}

@end

int main(int argc, const char *argv[]) {
    @autoreleasepool {
        Pizza *pizza = [[[[[PizzaBuilder alloc] init]
            setSize:@"Large"]
            addTopping:@"mushrooms"]
            addTopping:@"olives"]
            build];
        NSLog(@"%@", [pizza describe]);
    }
    return 0;
}
